app_dart/lib/src/service/agent_service.dart

```diff
@@ -0,0 +1,41 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert';
+import 'dart:math';
+
+import 'package:dbcrypt/dbcrypt.dart';
+import 'package:meta/meta.dart';
+
+/// Service class for Agent.
+///
+/// This service exists to provide common agent queries made by
+/// the Cocoon backend.
+@immutable
+class AgentService {
+  /// Generate new authorization token for [agent]
+  ///
+  /// The hashed code of token will be returned as a list
+  Map<String, dynamic> refreshAgentAuthToken() {
```

Comment:
Let's create an `AgentAuthToken` class that has a `value` getter for the raw String and a `hash` getter for the ascii-encoded hash - then return that instead of a map.

Reply:
Updated.